caddyhttp/proxy/policy.go

```diff
@@ -100,3 +103,30 @@ func (r *RoundRobin) Select(pool HostPool) *UpstreamHost {
 	}
 	return nil
 }
+
+// IPHash is a policy that selects hosts based on hashing the request ip
+type IPHash struct{}
+
+func hash(s string) uint32 {
+	h := fnv.New32a()
+	h.Write([]byte(s))
+	return h.Sum32()
+}
+
+// Select selects an up host from the pool using a round robin ordering scheme.
+func (r *IPHash) Select(pool HostPool, request *http.Request) *UpstreamHost {
+	poolLen := uint32(len(pool))
+	hash := hash(request.RemoteAddr)
```

Comment:
I was originally using `net.SplitHostPort()`  which looked like this.

```
clientIP, _, err := net.SplitHostPort(request.RemoteAddr);
if err != nil {
	return nil
}
hash := hash(clientIP)
```
However, `It's possible that the remote address will not have a port, in which case you would get a non-nil error`, was brought to my attention.  I'm happy to change it though.